packages/mui-base/package.json

```diff
@@ -40,6 +40,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.24.1",
+    "@floating-ui/react": "^0.26.10",
     "@floating-ui/react-dom": "^2.0.8",
```

Comment:
`@floating-ui/react` handles a ton of the complex logic we can re-use between floating components. Rewriting the work I did building this library between 2022-2023 is mostly a waste of time, especially due to the many edge cases it handles that would need to be re-implemented. By using it, I was able to get this Tooltip component almost completely ready in less than a day. Since it's designed to be wrapped in components like we're building in Base UI, it's a perfect fit (and fully tree-shakeable) 😄 . 

It's in v0, but I have full control of this library and plan to release v1 soon, before Base UI is v1. I'll try to get the APIs fully stable and learn from the floating components we're building to improve it.